rocAL/rocAL_pybind/amd/rocal/fn.py

```diff
@@ -590,3 +590,15 @@ def snp_noise(*inputs, snpNoise=None, device=None, preserve = False):
     kwargs_pybind = {"input_image0":inputs[0], "is_output":False ,"snpNoise": snpNoise}
     snp_noise_added_image = b.SnPNoise(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
     return (snp_noise_added_image)
+
+def external_source(*inputs, source, device=None, color_format=types.RGB, random_shuffle=False,mode=types.EXTSOURCE_FNAME, max_width=2000, max_heigth=2000):
+    # pybind call arguments
+    Pipeline._current_pipeline._external_source_operator = True
+    Pipeline._current_pipeline._external_source = iter(source)
+    Pipeline._current_pipeline._external_source_mode = mode
+    Pipeline._current_pipeline._external_source_user_given_width = max_width
+    Pipeline._current_pipeline._external_source_user_given_height = max_heigth
+    kwargs_pybind = {"source_path" : source.images_dir, "rocal_color_format" : color_format, "is_output" : False, "shuffle" : random_shuffle, "loop" : False, "decode_size_policy" : types.USER_GIVEN_SIZE,
+            "max_width" : 2000, "max_height" : 2000, "dec_type" : types.DECODER_TJPEG, "external_source_mode" : mode}
+    external_source_operator = b.ExternalFileSource(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
+    return (external_source_operator, []) # Labels is Empty
```

Comment:
Are we not passing labels?

Reply:
@swetha097 Please check this

Reply:
This is supposed to be empty, meta-data does nothing